docs/source-app/glossary/app_checkpointing.rst

```diff
@@ -0,0 +1,113 @@
+:orphan:
+
+#############
+Checkpointing
+#############
+
+Beta feature.
+
+----
+
+*********************
+What is a checkpoint?
+*********************
+When a lightning app is running it is possible to save the state of the app and start it again from the saved checkpoint.
+This is useful for long running apps that need to be stopped and restarted or while developing an app and needing to recreate the state of the app to to continue development at a specific point.
+
+
+************************
+How to save a checkpoint
+************************
+
+Saving a checkpoint is simple. In main flow if your lightning app, implement `should_save_checkpoint()` that returns True when you want to save a checkpoint.
+
+.. code-block:: python
+
+    class MyLightningApp(L.LightningFlow):
+        def run(self):
+            # do stuff
+            pass
+
+        def should_save_checkpoint(self):
+            return True
```

Comment:
Can any node of the flow tree define it or should we restrict it to the root only?

Reply:
Good question. It's only the root. I will add a comment here, and maybe enforce it with an error message in another PR.